### PR TITLE
CLOUDP-180570: add transformations to document experimental paths

### PR DESCRIPTION
**Pull request by matt-condon**
## Description

[CLOUDP-180570](https://jira.mongodb.org/browse/CLOUDP-180570): add transformations to document experimental paths

Since E2E validation has been performed on the new autogenerated Go SDK via AtlasCLI, we only have e2e validation for endpoints that are currently used by AtlasCLI

For that reason, we feel it is a necessary precaution to set "Experimental" tags on endpoints that do not have this level of validation, before going to GA

This PR prepends the descriptions of these endpoints endpoints with an "(experimental)" tag to denote that they are experimental 

### Proposed Solution:

Prepending the description of each experimental endpoint with "(experimental)" - based on the operationId mapped from the CLI-SDK coverage report (_operations.stable.json_)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by gssbzn — APPROVED** [review]

File: tools/transformer/src/transformations/tagExperimental.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @returns OpenAPI JSON File
+ */
+function applyAddExperimentalToDescriptions(api) {
+  Object.values(api.paths).forEach((path) => {
+    Object.values(path).forEach((requestMethod) => {
+      if (
+        requestMethod.operationId &&
+        !stableOperationIds.includes(requestMethod.operationId)
+      ) {
+        requestMethod.description = `(experimental) ${requestMethod.description}`;
```
Comment (wtrocki) [inline tools/transformer/src/transformations/tagExperimental.js:18]: [Nit] optional suggestion

```suggestion
        requestMethod.description = `[experimental]${requestMethod.description}`;
```

**Review by wtrocki — DISMISSED** [review]
LGTM. Thank you for the fix.

**Comment by wtrocki** [comment]
@jwilliams-mongo 

> question: is it trivial to also add it to https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_last_reference.md for each experimental method? I think that might be another place

It should be trivial. I think @matt-condon intention was to add this to the codebase that uses openapi `description`.
API docs uses `summary` as descriptions are overly long and sometimes contain html tags. 

We have 2 options:

1. Use OpenAPI summary field in the generated code (instead of the description). We already using OpenAPI summary in [doc_last_reference.md](https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_last_reference.md). Then individual documentation will be the only place where "description" is present

2. Add x-experimental to the OpenAPI and modify all the templates for code and docs to use it. 
This will allow us more control (we can have a separate table column or even use greyed-out color for those).

Both changes are trivial.

**Comment by wtrocki** [comment]
To support case 2: https://github.com/mongodb/atlas-sdk-go/pull/71
Case 1 seems not possible as we missing a summary for all open API paths for now. 
In the future we might consider moving to use summary only for the generated tools (leave descriptions for HTML content/docs)

File: docs/doc_1_concepts.md [hunk 2]
```diff
@@ -47,6 +47,10 @@ Use the `{Operation}()` method to perform modifications. For example:
  resp, r, err := sdk.ProjectsApi.CreateProjectInvitation(context.Background(), groupId, &groupInvitationRequest).Execute()
 ```
 
+## Experimental Tag
+
+Please note that some endpoints, denoted by the (experimental) tag in the operation description, have not undergone extensive end-to-end validation and may undergo changes in future releases.
+
 ## Example
```
Comment (wtrocki) [inline docs/doc_1_concepts.md:54]: I would rephrase it little bit to incluide:

- Flag that reason why we have those experiemental is that SDK is constantly updated with the changes
- Flag the risks: Feel free to use it but you might get breaking changes. Some inconsistencies might also appear. 
```suggestion
## Example
```
```suggestion
## Example
```
- Link to the reference containing all

File: docs/doc_1_concepts.md [hunk 3]
```diff
@@ -47,6 +47,10 @@ Use the `{Operation}()` method to perform modifications. For example:
  resp, r, err := sdk.ProjectsApi.CreateProjectInvitation(context.Background(), groupId, &groupInvitationRequest).Execute()
 ```
 
+## Experimental Tag
```
Comment (wtrocki) [inline docs/doc_1_concepts.md:50]: Experimental APIs? Experimental Methods?

File: docs/docs/AWSPrivateLinkConnection.md [hunk 4]
```diff
@@ -1,207 +0,0 @@
-# AWSPrivateLinkConnection
```
Comment (matt-condon) [inline docs/docs/AWSPrivateLinkConnection.md:1]: FYI doc regeneration job carried some additional changes from previous sdk generation where doc-generation was not run - changes not related - but fine to include in any case

**Review by jwilliams-mongo — APPROVED** [review]
@matt-condon LGTM thanks for making those tweaks. I think they'll make users happy.

File: tools/transformer/src/transformations/tagExperimental.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ */
+function applyAddExperimentalTag(api) {
+  Object.values(api.paths).forEach((path) => {
+    Object.values(path).forEach((requestMethod) => {
+      if (
+        requestMethod.operationId &&
+        !stableOperationIds.includes(requestMethod.operationId)
+      ) {
+        requestMethod.description = `${requestMethod.description}`;
+        requestMethod["x-experimental"] = true;
```
Comment (gssbzn) [inline tools/transformer/src/transformations/tagExperimental.js:17]: not blocking, should this be x-xgen-experimental?

File: tools/config/go-templates/README.mustache [hunk 6]
```diff
@@ -4,9 +4,9 @@
 
 All URIs are relative to *{{basePath}}*
 
-Class | Method | HTTP request | Description
------------- | ------------- | ------------- | -------------
-{{#apiInfo}}{{#apis}}{{#operations}}{{#operation}}*{{classname}}* | [{{operationId}}](./{{apiDocPath}}{{classname}}.md#{{operationIdLowerCase}}) | **{{httpMethod}}** {{path}} | {{summary}}
+Class        | Method        | HTTP request  | Description   | Experimental
```
Comment (gssbzn) [inline tools/config/go-templates/README.mustache:7]: like discussed on the call can we make the tittle of the table, "Stability level" and instead of yes no, make it Stable/Experimental?

File: tools/transformer/src/transformations/tagExperimental.js (hunk 5 above)
Comment (matt-condon) [reply tools/transformer/src/transformations/tagExperimental.js]: Agreed, updated

File: tools/config/go-templates/README.mustache (hunk 6 above)
Comment (matt-condon) [reply tools/config/go-templates/README.mustache]: Done

**Review by gssbzn — APPROVED** [review]
LGTM 💯

File: tools/config/go-templates/README.mustache (hunk 6 above)
Comment (wtrocki) [reply tools/config/go-templates/README.mustache]: Experimental means = we have that untested in SDK. "Stability level" is not documented term and likely to mean API stability levels in the future. We can imagine having alpha stability level being experimental and also Production also being untested.

**Comment by wtrocki** [comment]
Merging as I will get a lot of conflicts on other PRs.
